search.go

```diff
@@ -353,6 +353,46 @@ func (t *Trie) search(prefixKey []string, opts *SearchOptions) *SearchResults {
 	return results
 }
 
+// SelectOnValue: find item based on value.
+// apply eval func to each Value of the tree
+// build a result with key, value for each itme eval == true
+func (t *Trie) SelectOnValue(eval func(interface{}) bool) *SearchResults {
+	results := &SearchResults{}
+	node := t.root
+	t.selectOnValue(results, node, []string{}, eval)
+	return results
+}
+
+// selectOnValue: recursively apply eval an item tree
+func (t *Trie) selectOnValue(results *SearchResults, node *Node, prefixKey []string, eval func(interface{}) bool) *SearchResults {
+
+	for key, child := range node.children {
```

Comment:
Use childrenDLL to give it a deterministic walk-order. For reference see: https://github.com/shivamMg/trie/blob/master/search.go#L196
![image](https://user-images.githubusercontent.com/7882752/223314163-6317c580-e226-477d-ae42-ba2556e6a691.png)

Idea is to iterate through children using a doubly-linked list instead of map keys which is not deterministic

Reply:
I wonder, why you where using childrenDLL and children attributes.
I have renamed the function walk, and change the implementation tu use the childrenDLL. test are updated to.